cg/services/bcl_convert_metrics_service/bcl_convert_metrics_service.py

```diff
@@ -111,3 +111,26 @@ def create_sample_run_metrics(
             base_mean_quality_score=mean_quality_score,
             created_at=datetime.now(),
         )
+
+    def create_sample_sequencing_metrics_for_flow_cell(
+        self,
+        flow_cell_directory: Path,
+        run_metrics_id: int,
+        store: Store,
+    ) -> list[IlluminaSampleSequencingMetrics]:
+        """Parse the demultiplexing metrics data into the sequencing statistics model."""
+        metrics_parser = MetricsParser(flow_cell_directory)
+        sample_internal_ids: list[str] = metrics_parser.get_sample_internal_ids()
+        sample_lane_sequencing_metrics: list[IlluminaSampleSequencingMetrics] = []
+
+        for sample_internal_id in sample_internal_ids:
+            for lane in metrics_parser.get_lanes_for_sample(sample_internal_id=sample_internal_id):
+                metrics: IlluminaSampleSequencingMetrics = self.create_sample_run_metrics(
+                    sample_internal_id=sample_internal_id,
+                    lane=lane,
+                    metrics_parser=metrics_parser,
+                    run_metrics_id=run_metrics_id,
+                    store=store,
+                )
+                sample_lane_sequencing_metrics.append(metrics)
```

Comment:
Could this be extracted into a function? Something like `create_metrics_for_sample`. And then loop trough the lanes in that function?

Reply:
Will adress this in next PR